src/main/java/com/auctionapp/api/repository/specification/GenericSpecificationsBuilder.java

```diff
@@ -0,0 +1,57 @@
+package com.auctionapp.api.repository.specification;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import org.springframework.data.jpa.domain.Specification;
+
+public class GenericSpecificationsBuilder<T> {
+   
+	private final List<SearchCriteria> params;
+	private final List<Specification<T>> specifications;
+	
+	public GenericSpecificationsBuilder() {
+	   this.params = new ArrayList<>();
+	   this.specifications = new ArrayList<>();
+	}
+ 
+	public final GenericSpecificationsBuilder<T> with(final String key, 
+													  final SearchOperation searchOperation, 
+													  final List<Object> arguments) {
+	   return with(key, searchOperation, false, arguments);
+	}
+	
+	public final GenericSpecificationsBuilder<T> with(final String key, 
+													  final SearchOperation searchOperation, 
+													  final boolean isOrOperation, 
+													  final List<Object> arguments) {
+	   params.add(new SearchCriteria(key, searchOperation, isOrOperation, arguments));
+	   return this;
+	}
+	public final GenericSpecificationsBuilder<T> with(final Specification<T> specification) {
+	   specifications.add(specification);
+	   return this;
+	}
+	public Specification<T> build() {
+	   Specification<T> result = null;
+	   if (!params.isEmpty()) {
+		  result = new GenericSpecification<>(params.get(0));
+		  for (int index = 1; index < params.size(); ++index) {
+			 SearchCriteria searchCriteria = params.get(index);
+			 result = searchCriteria.isOrOperation() ? Specification.where(result).or(new GenericSpecification<>(searchCriteria))
+				: Specification.where(result).and(new GenericSpecification<>(searchCriteria));
```

Comment:
It's better to write out a normal if condition since the condition with the ternary operator cannot fit within one line and is, therefore, less readable. Clarity > brevity.